packages/shared/src/lib/core/layer-2/actions/generateAndStoreEvmAddressForAccounts.ts

```diff
@@ -0,0 +1,40 @@
+import { IAccountState } from '@core/account/interfaces'
+import { Ledger } from '@core/ledger/classes'
+import { ProfileType } from '@core/profile/enums'
+import { api, getProfileManager } from '@core/profile-manager'
+import { updateActiveAccountPersistedData } from '@core/profile/actions'
+import { updateActiveAccount } from '@core/profile/stores'
+
+export async function generateAndStoreEvmAddressForAccounts(
+    profileType: ProfileType,
+    coinType: number,
+    ...accounts: IAccountState[]
+): Promise<void> {
+    if (accounts.length > 0 && accounts[0].evmAddresses?.[coinType]) {
+        return
+    }
+
+    for (const account of accounts) {
+        const accountIndex = account.getMetadata().index
```

Comment:
```suggestion
        const accountIndex = account.index
```
although i think we dont need to declare this as a variable

Reply:
It's best to call the function because when logging I noticed that the non-function properties weren't always there, b/c of the different with `IAccount` and `IAccountState`